omegaconf/dictconfig.py

```diff
@@ -199,7 +211,7 @@ def _validate_set(self, key: Any, value: Any) -> None:
             and not issubclass(value_type, target_type)
         )
         if validation_error:
-            self._raise_invalid_value(value, value_type, target_type)
+            _raise_invalid_assignment(target_type, value_type, value)
```

Comment:
Is this a fix to https://github.com/facebookresearch/hydra/issues/1322 by any chance?

Reply:
No, in dictconfig.py:629 
```python
else:  # pragma: no cover
    msg = f"Unsupported value type : {value}"
    raise ValidationError(msg)
```
should print type_str(type(value)). Should I fix it in this PR?

Reply:
I am not sure it's that simple. we can do it in a different pr (need a clean standalone repro first).